packages/block-editor/src/components/link-control/use-create-page.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+import { useEffect, useState, useRef } from '@wordpress/element';
+
+export default function useCreatePage( customCreatePage = null ) {
+	const { saveEntityRecord } = useDispatch( 'core' );
```

Comment:
This can't be used in "block-editor". Anything that has direct relationship with WP API can't be used as the block-editor package is a client only package agnostic to WP context.

Reply:
@youknowriad what do you think would be the right package for this code to live? I can't quite fit this into any existing one. Are these two hooks (`useSearchHandler` and `useCreatePage`) enough to justify creating a new package?

Reply:
It seems like `useSearchHandler` still fits the block-library as it depends on the editor setting, not on any WP-specific logic. It seems like the same trick should work for the `useCreatePage` - let me try it.

Reply:
It would work but ATM the go-to solution for these things is copying and pasting the code to every package where the editor is initialized. I don't like that so I went back to passing the logic from the navigation-block/edit which ATM is the only consumer of that feature.